### PR TITLE
[fix]: [PIE-5003]: NPE is thrown while saving a pipeline in git-sync(Old Git-sync method is invoked, while the FF for old git-sync is not enabled)

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes: 
Issue: 
Pipeline save is failing in our old git sync pipeline automation probably because the account does not have the FF for old git sync enabled, and hence old sync is not being enabled for any project. If you go to the project, you’ll not see any git management tab anymore.
Changes: 
StoreType will have default value INLINE from now onwards and also Added a check for if StoreType is null, then it will be set to INLINE(By Default).

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36710)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36710)

File: 970-ng-commons/src/main/java/io/harness/gitsync/beans/StoreType.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public static StoreType getFromStringOrNull(String value) {
     try {
```
Comment (namanvermah) [inline 970-ng-commons/src/main/java/io/harness/gitsync/beans/StoreType.java:18]: change method name to "getFromStringOrInline"

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36710)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`970-ng-commons/src/main/java/io/harness/gitsync/beans/StoreType.java` line 18 at r2](https://reviewable.io/reviews/harness/harness-core/36710#-NADewT1U1tYgSpBkZZy-r2-18:-NAECDDmEgFbrKcCAL_7:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/88fcf55d385bf05e15ca7826f0e4cf8ee8888912/970-ng-commons/src/main/java/io/harness/gitsync/beans/StoreType.java#L18)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

change method name to "getFromStringOrInline"
</blockquote></details>

Done.

**Comment by senjucanon2** [comment]
trigger buildnum

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36710#-:-NAIHlcA8XwzQgb_AHXu:bnfp4nl)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger messagecheck

**Review by naidu-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36710#-:-NAP5nFr874VgppC4IHK:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**